# OrthologMotif User Guide

1.  ## Overview

OrthologMotif is a software program written in the C\# language for the
purpose of analyzing motif-containing orthologous proteins in multiple
species. You can find the code at
<https://github.com/moli-lab/OrthologMotif/releases> and this software
provides users with console application and WPF desktop program.

2.  ## WPF desktop application

    ### download
    
    Download and unzip OrthologMotif.zip, then click
    motif-confirm-mouseClick.exe to access the program. It provides a
    species library containing 150 species, with numbers in parentheses
    indicating evolutionary distances of each species and two input boxes,
    allowing users to input motif or load motif file.
    
    ### Input and Usage
    
    Query species, target species and motif are three essential elements of
    software operation. The desktop program provides a user interface for
    visualization operations. Select the corresponding species and click the
    *Add* or *Remove* button next to the two input boxes to add or remove
    the selected species to the query species or target species,
    respectively. In terms of motif, it can be loaded from a file or entered
    in the input box. After selecting corresponding species as query and
    target species and load the file of motif, click *Start* button to
    launch the program. During the performing, users can minimize window by
    right-clicking.
    
3. ## Console application

   

   ### Download and Setup

   Download and unzip OrthologMotif.zip, then open the terminal and change
   into directory to current folder. Running the *motif-confirm.exe* *-h*
   command to obtain the usage and description of the software.

   Input and Usage

> OrthologMotif handles the following input parameters:
>
> *-query* (required): The list of selected query species, separated by
> space.
>
> *-subject* (required): The selected target species. Only one is allowed.
> If multiple species are given, the first is accepted by default.

> *-motif* (required): The path of the motif file
>
> *-h* : The usage and description of the software. Once accept this
> parameter, ignore all other parameters by default.

Set the query species, target species and motif through the command line
interface and press *Enter*. If the given command contains error
parameter, the program would echo ERROR and error messages, otherwise
echo parameters information for confirmation. Press *n* to terminate the
program the operation, enter any other key to continue.

## Output

After the run completes, three files would be written to result file.
The following files will be generated by OrthologMotif:

\<date\>\#containMotifNum\#{target species}.txt: Table of theoretical
and actual values of each input motif in each query species.

\<date\>\#proteinID\#{target species}.txt: List of the input
motif-containing orthologous protein IDs, which can be used to track
orthologous proteins.

\<date\>\#table\#{target species}.txt: List of the number of
motif-containing orthologous proteins present in each query species. 

4. ## sample

Here, an example is used to illustrate how to use this software. In this
example, the target species is hsap(H. sapiens) and query species are
crei(Chlamydomonas reinhardtii), cbri(Caenorhabditis briggsae AF16),
cele(Caenorhabditis elegans), ncan(Neospora caninum), isca(Ixodes
scapularis), nvec(Nematostella vectensis), aaeg(Aedes aegypti),
dmel(Drosophila melanogaster), agam(Anopheles gambiae str. PEST),
cint(Ciona intestinalis), apis(Acyrthosiphon pisum), drer(Danio rerio),
tnig(Tetraodon nigroviridis), trub(Takifugu rubripes),ggal(Gallus
gallus), rnor(Rattus norvegicus), mmus(Mus musculus), clup(Canis lupus
familiaris), ecab(Equus caballus), ptro(Pan troglodytes). Motifs are
amino acid sequences with the pattern A\*\*A\*\*A\*\*A\*\*A (5A, A is an
any amino acid such as L, E, Q etc.)

1.  **WPF desktop application**
1.  **Query species**

> Select the species listed above and the click *Add* button on the
> left, and these selected species will be listed in the box of Query
> Species on the left.

![image-20200216155216759](img/Query_species.png)

2.  **Target species**

> Select the *hsap* and click the *Add* button on the right, and it will
> be listed in the box of Subject Species on the right.

![image-20200216155236123](img/Target_species.png)

3.  **Motif**

Click the *Open* button and select the motif file so that the motifs
will be loaded in the Motif Box. User can also enter motifs in the Motif
Box. After setting all parameters, click *Start* button to launch the
program.

![image-20200216155301018](img/Motif_1.png)

![image-20200216155311401](img/Motif_2.png)

2.  **Console Application**

Open the terminal and change directiry to current folder. Running the
*-motif-confirm.exe -query crei cbri cele ncan isca ncec aaeg dmel agam
cint apis drer tnig rnor mmus clup ecab ptro -subject hsap -motif
motif.txt* command and click any key other than *n* to launch the
program**.**

![image-20200216155323256](img/Console_Application.png)
